src/stream/src/executor/managed_state/dynamic_filter.rs

```diff
@@ -196,6 +197,7 @@ impl<S: StateStore> RangeCache<S> {
             }
         }
         self.vnodes = new_vnodes;
+        self.range = None;
```

Comment:
As discussed in https://github.com/risingwavelabs/risingwave/issues/6414, can we reset the range and the cache iff `new_vnodes` is not a subset of `old_vnodes`, else delete the stale vnodes if it is? I believe there is a helper function to check this https://github.com/risingwavelabs/risingwave/blob/c26aa3cc35be73784d64704903245c46639a1e75/src/stream/src/cache/mod.rs#L118.

Reply:
Yes. This can be a quick but large improvement.